src/main/java/duchess/task/Event.java

```diff
@@ -0,0 +1,103 @@
+
+    /** The DateTimeFormatter when converting time from string.*/
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy ha");
+
+    /** The DateTimeFormatter when converting time with minutes from string.*/
+    private static final DateTimeFormatter DATE_MINUTES_FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy h:mma");
+
+
+    /**
+     * Constructs an Event.
+     * @param name The name of the event.
+     * @param dateTimeStart The time which the event starts.
+     * @param dateTimeEnd The time which the event ends.
+     */
+    public Event(String name, LocalDateTime dateTimeStart, LocalDateTime dateTimeEnd) {
+        super(name);
+        this.dateTimeStart = dateTimeStart;
+        this.dateTimeEnd = dateTimeEnd;
+    }
+
+    /**
+     * Converts the user input string to a LocalDateTime.
+     * @param date The user input date as a string.
+     * @param duration The user input duration as a string.
+     * @return An array of LocalDateTime containing the start and end LocalDateTimes respectively.
+     * @throws DuchessException When an incorrect format is used for Event.
+     */
+    public static LocalDateTime[] convertStringToDate (String date, String duration) throws DuchessException {
+        try {
+            String replacement = duration.replace("am", "AM").replace("pm","PM");
+            String[] timeParts = replacement.split("-");
+            String startTime = timeParts[0];
+            String endTime = timeParts[1];
+            LocalDateTime startEvent = LocalDateTime.parse(date + " " + startTime,
+                    startTime.contains(":") ? DATE_MINUTES_FORMATTER : DATE_FORMATTER);
+            LocalDateTime endEvent = LocalDateTime.parse(date + " " + endTime,
+                    endTime.contains(":") ? DATE_MINUTES_FORMATTER : DATE_FORMATTER);
+            LocalDateTime[] localDateTimes = {startEvent, endEvent};
+            return localDateTimes;
+        } catch (DateTimeException e) {
+            throw new DuchessException("Wrong format used.");
+        }
+    }
+
+    public LocalDateTime getDateTime(){
+        return this.dateTimeStart;
+    }
+
+    /**
+     * Returns a simplified representation of the Event for easier recovery from save file.
+     * @return The file formatted string representation of the Event.
+     */
+    public String toFileFormat() {
+        return String.format("E%s,%s,%s,%s", name, dateTimeStart.format(PRINT_DATE_FORMATTER),
+                dateTimeEnd.format(PRINT_DATE_FORMATTER), isDone);
+    }
+
+    /**
+     * Converts the user input string to a LocalDateTime.
+     * @param by The date as text within the file.
+     * @return The LocalDateTime representation.
+     * @throws DuchessException Exception thrown when an incorrect format is used for Deadline.
+     */
+    public static LocalDateTime convertTextToDate (String by)  {
+        return LocalDateTime.parse(by, PRINT_DATE_FORMATTER);
+    }
```

Comment:
Should there just be a single space between the method name and the curly bracket? Line 70 there is no space and line 89 there are two spaces
I also noticed this in your other child Task classes